backend/src/main/java/ca/mcgill/purposeful/service/CollaborationResponseService.java

```diff
@@ -0,0 +1,54 @@
+package ca.mcgill.purposeful.service;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+
+import ca.mcgill.purposeful.dao.CollaborationRequestRepository;
+import ca.mcgill.purposeful.dao.CollaborationResponseRepository;
+import ca.mcgill.purposeful.exception.GlobalException;
+import ca.mcgill.purposeful.model.CollaborationRequest;
+import ca.mcgill.purposeful.model.CollaborationResponse;
+import ca.mcgill.purposeful.model.Idea;
+import jakarta.transaction.Transactional;
+
+/**
+ * Collaboration response service functions
+ */
+public class CollaborationResponseService {
+    @Autowired
+    CollaborationResponseRepository collaborationResponseRepository;
+
+    @Autowired
+    CollaborationRequestRepository collaborationRequestRepository;
+
+    @Autowired
+    IdeaService ideaService;
+
+    /**
+     * Get the collaboration response for a given idea
+     * 
+     * @param ideaUuid the UUID of the idea
+     * @return the collaboration response (or null if there is none)
+     * @throws GlobalException if the idea does not have a collaboration request from the user
+     * 
+     * @author Thibaut Baguette
+     */
+    @Transactional
+    public CollaborationResponse getCollaborationResponseForIdea(String ideaUuid) {
```

Comment:
So I think this may not work if I understand it correctly. I think its possible that one idea has many requests from many different users, so I think the service method would also need to know the email of the person who is looking for the collaboration response. That way, we get all the requests for the idea, find the request that belongs to a particular user, and if that user has sent a request, check if there is a response. Return null if there is no response. Return error if they have not sent a request. What do you think?

Reply:
Yeah I hadn't thought of that. I'll change it